chain/chain/src/chain.rs

```diff
@@ -2861,10 +2872,16 @@ impl<'a> ChainUpdate<'a> {
             return Err(ErrorKind::Orphan.into());
         }
 
+        let block_height = block.header().height();
         // A heuristic to prevent block height to jump too fast towards BlockHeight::max and cause
         // overflow-related problems
-        if block.header().height() > head.height + self.epoch_length * 20 {
-            return Err(ErrorKind::InvalidBlockHeight.into());
+        if block_height > head.height + self.epoch_length * 20 {
+            return Err(ErrorKind::InvalidBlockHeight(block_height).into());
+        }
+
+        // Do not accept blocks that are too old.
+        if block_height < self.runtime_adapter.get_gc_stop_height(&head.last_block_hash)? {
```

Comment:
This is not correct. We need to make sure that Height of the Prev Block > get_gc_stop_height. The reason is, we want to forbid building Blocks on the top of old forks.